File: Data Structures & Algorithms/majority-element/submission-3.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = 0 , n = nums.size();
        for(int i = 0 ; i < 32 ; i++){
            int count = 0;
            for(int j = 0 ; j<n ; j++){
                if((1<<i) & (nums[j]))count++;
            }
            cout<<count<<" "<<(1<<i)<<endl;
            if(count > (n / 2)) ans = ans | (1<<i);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l=0
        r=len(nums)-1
        while l<r:
            mid=l+(r-l)//2
            if nums[mid]<nums[r]:
                r=mid
            else:
                l=mid+1
        ls=nums[l:]+nums[0:l]
        left=0
        right=len(ls)-1
        c=-1
        while left<=right:
            mid=left+(right-left)//2
            if ls[mid]<target:
                left=mid+1
            elif ls[mid]>target:
                right=mid-1
            else:
                c=mid
                break
        print(ls)
        if c==-1:
            return -1
        else:
            return (c+l)%len(nums)
